tokio/src/util/mod.rs

```diff
@@ -19,3 +19,5 @@ cfg_rt_threaded! {
 #[cfg(any(feature = "macros", feature = "stream"))]
 #[cfg_attr(not(feature = "macros"), allow(unreachable_pub))]
 pub use rand::thread_rng_n;
+
+pub(crate) mod intrusive_double_linked_list;
```

Comment:
I think this branch needs to be rebased onto master & updated to use the existing doubly-linked list implementation? Or are there missing features in the current implementation that this branch needs?

Reply:
Yeah, it probably would need a rebase to do this. I'm not sure whether something is missing. I might actually have some preferrence for cleaning the 2 lists up in a follow-up CR, since I wouldn't like to list stuff again with everything else. Fixed enough bugs in the current CR regarding lists and locking :-)